library/std/src/sys/unix/ext/ucred.rs

```diff
@@ -0,0 +1,93 @@
+//! Unix peer credentials.
+
+// NOTE: Code in this file is heavily based on work done in PR 13 from the tokio-uds repository on
+//       GitHub.
+//
+//       For reference, the link is here: https://github.com/tokio-rs/tokio-uds/pull/13
+//       Credit to Martin Habovštiak (GitHub username Kixunil) and contributors for this work.
+
+use libc::{gid_t, uid_t};
+
+/// Credentials for a UNIX process for credentials passing.
+#[unstable(feature = "peer_credentials_unix_socket", issue = "42839", reason = "unstable")]
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct UCred {
+    pub uid: uid_t,
+    pub gid: gid_t,
```

Comment:
`something_t` types are not used in `std` often (at all?), is this a good idea?

Personally I'd love seeing use of newtypes as done in `nix`, but not sure if that should go to `std`.

Reply:
I did a quick `grep` of the codebase and I can see a few places where these are used... I'll leave it as it is for now! If you would strongly like this to change feel free to follow up. 🙂